From 83eab80fcdaec9e9b6744810ca205fdafe4b1663 Mon Sep 17 00:00:00 2001
From: Stefano Duo <stefanoduo@google.com>
Date: Wed, 11 Jan 2023 11:56:51 +0000
Subject: [PATCH] Introduce is_cronet_build and move process_launcher into its
 own target

Cronet doesn't require the setup and handling of multiple processes
(it simply spawns threads within the application process that uses
it). This wouldn't be that big of a problem, but many of the
multiprocess tooling makes use of AndroidX, which Cronet is trying not
to depend on.

To achieve that, move process_launcher files into its own target not
included in base_java (which Cronet depends on). Also, drop
IntentUtils for Cronet's builds (through the use of the new GN arg
is_cronet_built).
---
 base/BUILD.gn               | 53 ++++++++++++++++++++++++++-----------
 base/test/BUILD.gn          |  1 +
 build/config/ios/config.gni |  5 ----
 net/BUILD.gn                |  2 +-
 4 files changed, 39 insertions(+), 22 deletions(-)

diff --git a/base/BUILD.gn b/base/BUILD.gn
index 5062dd8260d19..c2263d7a48f91 100644
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -25,6 +25,7 @@ import("//build/config/c++/c++.gni")
 import("//build/config/chromecast_build.gni")
 import("//build/config/chromeos/ui_mode.gni")
 import("//build/config/compiler/compiler.gni")
+import("//build/config/cronet/config.gni")
 import("//build/config/dcheck_always_on.gni")
 import("//build/config/ios/config.gni")
 import("//build/config/logging.gni")
@@ -4163,10 +4164,39 @@ if (is_android) {
     template = "android/java/src/org/chromium/base/BaseSwitches.java.tmpl"
   }
 
+  android_library("process_launcher_java") {
+    srcjar_deps = [ ":process_launcher_aidl" ]
+    deps = [
+      ":base_java",
+      ":jni_java",
+      "//build/android:build_java",
+      "//third_party/android_deps:com_google_code_findbugs_jsr305_java",
+      "//third_party/androidx:androidx_annotation_annotation_experimental_java",
+      "//third_party/androidx:androidx_annotation_annotation_java",
+      "//third_party/androidx:androidx_collection_collection_java",
+      "//third_party/androidx:androidx_core_core_java",
+    ]
+    sources = [
+      "android/java/src/org/chromium/base/process_launcher/BindService.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildConnectionAllocator.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildProcessConnection.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildProcessConstants.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildProcessLauncher.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildProcessService.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildProcessServiceDelegate.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnection.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnectionDelegate.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnectionFactory.java",
+      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnectionImpl.java",
+      "android/java/src/org/chromium/base/process_launcher/FileDescriptorInfo.java",
+    ]
+
+    annotation_processor_deps = [ "//base/android/jni_generator:jni_processor" ]
+  }
+
   android_library("base_java") {
     srcjar_deps = [
       ":base_android_java_enums_srcjar",
-      ":base_java_aidl",
       ":java_features_srcjar",
       ":java_switches_srcjar",
     ]
@@ -4209,7 +4239,6 @@ if (is_android) {
       "android/java/src/org/chromium/base/Function.java",
       "android/java/src/org/chromium/base/ImportantFileWriterAndroid.java",
       "android/java/src/org/chromium/base/IntStringCallback.java",
-      "android/java/src/org/chromium/base/IntentUtils.java",
       "android/java/src/org/chromium/base/JNIUtils.java",
       "android/java/src/org/chromium/base/JavaExceptionReporter.java",
       "android/java/src/org/chromium/base/JavaHandlerThread.java",
@@ -4291,18 +4320,6 @@ if (is_android) {
       "android/java/src/org/chromium/base/metrics/TimingMetric.java",
       "android/java/src/org/chromium/base/metrics/UmaRecorder.java",
       "android/java/src/org/chromium/base/metrics/UmaRecorderHolder.java",
-      "android/java/src/org/chromium/base/process_launcher/BindService.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildConnectionAllocator.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildProcessConnection.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildProcessConstants.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildProcessLauncher.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildProcessService.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildProcessServiceDelegate.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnection.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnectionDelegate.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnectionFactory.java",
-      "android/java/src/org/chromium/base/process_launcher/ChildServiceConnectionImpl.java",
-      "android/java/src/org/chromium/base/process_launcher/FileDescriptorInfo.java",
       "android/java/src/org/chromium/base/supplier/BooleanSupplier.java",
       "android/java/src/org/chromium/base/supplier/DestroyableObservableSupplier.java",
       "android/java/src/org/chromium/base/supplier/ObservableSupplier.java",
@@ -4331,6 +4348,10 @@ if (is_android) {
       "android/java/src/org/chromium/base/task/TaskTraitsExtensionDescriptor.java",
     ]
 
+    if (!is_cronet_build) {
+      sources += [ "android/java/src/org/chromium/base/IntentUtils.java" ]
+    }
+
     annotation_processor_deps = [ "//base/android/jni_generator:jni_processor" ]
 
     resources_package = "org.chromium.base"
@@ -4342,7 +4363,7 @@ if (is_android) {
     ]
   }
 
-  android_aidl("base_java_aidl") {
+  android_aidl("process_launcher_aidl") {
     import_include = [ "android/java/src" ]
     sources = [
       "android/java/src/org/chromium/base/process_launcher/IChildProcessService.aidl",
@@ -4517,7 +4538,7 @@ if (is_android) {
   android_library("base_java_process_launcher_test_support") {
     testonly = true
 
-    deps = [ ":base_java" ]
+    public_deps = [ ":process_launcher_java" ]
 
     sources = [ "test/android/javatests/src/org/chromium/base/process_launcher/TestChildProcessConnection.java" ]
   }
diff --git a/base/test/BUILD.gn b/base/test/BUILD.gn
index 96149187cb5d7..f3bdead3dc7a3 100644
--- a/base/test/BUILD.gn
+++ b/base/test/BUILD.gn
@@ -393,6 +393,7 @@ if (is_android) {
     deps = [
       "//base:base_java",
       "//base:jni_java",
+      "//base:process_launcher_java",
       "//testing/android/native_test:native_main_runner_java",
       "//third_party/android_deps:com_google_code_findbugs_jsr305_java",
     ]
diff --git a/build/config/ios/config.gni b/build/config/ios/config.gni
index 74d064532d5f8..c5c10c3f7e5c3 100644
--- a/build/config/ios/config.gni
+++ b/build/config/ios/config.gni
@@ -9,11 +9,6 @@ declare_args() {
   # default is only there for compatibility reasons and will be removed (see
   # crbug.com/1138425 for more details).
   target_environment = ""
-
-  # Control whether cronet is built (this is usually set by the script
-  # components/cronet/tools/cr_cronet.py as cronet requires specific
-  # gn args to build correctly).
-  is_cronet_build = false
 }
 
 if (target_environment == "") {
diff --git a/net/BUILD.gn b/net/BUILD.gn
index e12e5b6936b8d..53f8ae99f62d2 100644
--- a/net/BUILD.gn
+++ b/net/BUILD.gn
@@ -6,8 +6,8 @@ import("//build/buildflag_header.gni")
 import("//build/config/chromecast_build.gni")
 import("//build/config/chromeos/ui_mode.gni")
 import("//build/config/compiler/compiler.gni")
+import("//build/config/cronet/config.gni")
 import("//build/config/features.gni")
-import("//build/config/ios/config.gni")
 import("//crypto/features.gni")
 import("//net/features.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
-- 
2.39.0.314.g84b9a713c41-goog

